app/models/salary_graph.rb

```diff
@@ -8,30 +8,17 @@ def initialize(employees, dates)
 
   def to_table
     dates.map do |date|
-      # Multiply by 1000 to convert from seconds since the epoch to
-      # milliseconds since the epoch
-      format_date(date) + employees_row_for(date)
+      employees_row_for({ date: date })
     end
   end
 
   private
 
-  def format_date(date)
-    [date.to_time.to_f * 1000]
-  end
-
-  def employees_row_for(date)
-    employees.map { |e| [e.salary_on(date), tooltip_for_date(e, date)] }.flatten
-  end
-
-  def tooltip_for_date(employee, date)
-    "#{date}\n#{employee.display_name}: #{format_salary(employee.salary_on(date))}"
-  end
-
-  def format_salary(salary)
-    if salary
-      salary_in_ks = salary / 1000
-      "$#{format('%g', salary_in_ks)}K"
+  def employees_row_for(one)
```

Comment:
Sorry, that was a bad name I picked.  How about something like `chart_data = { date: date } ` and inside the each `chart_data[key] = { name: e.display_name, salary: e.salary_on(date) }` ?